pkg/volume/aws_ebs/aws_util.go

```diff
@@ -129,6 +132,23 @@ func (util *AWSDiskUtil) CreateVolume(c *awsElasticBlockStoreProvisioner) (aws.K
 		return "", 0, nil, "", fmt.Errorf("both zone and zones StorageClass parameters must not be used at the same time")
 	}
 
+	if utilfeature.DefaultFeatureGate.Enabled(features.DynamicProvisioningScheduling) && node != nil {
+		if node.ObjectMeta.Labels == nil {
+			return "", 0, nil, "", fmt.Errorf("labels for node missing")
+		}
+		z, ok := node.ObjectMeta.Labels[kubeletapis.LabelZoneFailureDomain]
+		if !ok {
+			return "", 0, nil, "", fmt.Errorf("%s Label for node missing", kubeletapis.LabelZoneFailureDomain)
+		}
+
+		if volumeOptions.ZonePresent || volumeOptions.ZonesPresent {
+			return "", 0, nil, "", fmt.Errorf("zone/zones cannot be specified with DynamicProvisioningScheduling enabled")
```

Comment:
Maybe you can suggest to use allowedTopologies instead

Reply:
or `zone/zones cannot be specified with VolumeBindingMode = WaitForFirstConsumer`.

Reply:
@msau42 , from what I understand, with `DynamicProvisioningScheduling` enabled, `AllowedTopologies` will only be relevant for scenarios like GCE where multi-zonal volumes are supported, correct? In case of EBS, I am ignoring `AllowedTopologies`. I will update the error msg to what Jan suggested above.

Reply:
No, AllowedTopologies is relevant to single-zone volumes too.  It basically replaces the "zone" and "zones" parameters, and allows an admin to restrict the zones where volumes can be provisioned in.  The difference is the scheduler will respect it when choosing a node.

Reply:
Here's what I think the provision implementation of single-zone volumes should do:
1. node not set, allowed topologies not set: old behavior
2. node not set, allowed topologies set: choose from allowed topologies
3. node set, zone/zones set: error always?
4. node set, allowed topologies set or not set: use node's zone
5. zone/zones set + allowedtopologies set: error